graphql_api/query/personal_access_token.py

```diff
@@ -0,0 +1,33 @@
+import graphene
+
+from controller.auth import manager as auth
+from controller.personal_access_token import manager as token_manager
+from graphql_api.types import PersonalAccessToken
+
+
+class PersonalAccessTokenQuery(graphene.ObjectType):
+
+    personal_access_token = graphene.Field(
+        PersonalAccessToken,
+        project_id=graphene.ID(required=True),
+        name=graphene.String(required=True),
+    )
+
+    all_personal_access_tokens = graphene.Field(
+        graphene.List(PersonalAccessToken), project_id=graphene.ID(required=True)
+    )
+
+    def resolve_personal_access_token(
+        self, info, project_id: str, name: str
+    ) -> PersonalAccessToken:
+        auth.check_demo_access(info)
+        auth.check_project_access(info, project_id)
+        user_id = auth.get_user_id_by_info(info)
+        return token_manager.get_personal_access_token(project_id, user_id, name)
+
+    def resolve_all_personal_access_tokens(
+        self, info, project_id: str
+    ) -> PersonalAccessToken:
+        auth.check_demo_access(info)
+        auth.check_project_access(info, project_id)
```

Comment:
i think we need to check for admin access as well here tough I'm not sure if we want to do this now or later once we have the actual admin role

Reply:
Added for PAT queries and mutations "auth.check_admin_access(info)" so that only kern admins can access these endpoints, as soon as we introduce the admin role we can change it to however the design is then.